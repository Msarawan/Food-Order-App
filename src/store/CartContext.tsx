import React, { useContext } from 'react';

const CartContext = React.createContext({
  items: [],
  totalAmount: 0,
  addItem: (item:any) => {},
  removeItem: (id:String) => {}
});

export const CartCtx = () => useContext(CartContext);
export default CartContext;