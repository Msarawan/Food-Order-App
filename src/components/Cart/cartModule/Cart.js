// { useContext } from 'react';
import Modal from '../../UI/modal/Modal';
import CartItem from '../cartItem/CartItem';
import classes from './Cart.module.css';
//import CartContext from '../../../store/CartContext';
import { CartCtx } from '../../../store/CartContext';

const Cart = (props) => {
  //const cartCtx = useContext(CartContext);
  const cartCtx = CartCtx();
  const totalAmount = `$${cartCtx.totalAmount?.toFixed(2)}`;

  const cartItemRemoveHandler = (id) => {
    cartCtx.removeItem(id);
  };

  const cartItemAddHandler = (item) => {
    cartCtx.addItem({ ...item, amount: 1 });
};
   
 console.log(cartCtx.items);

const cartItems = (
    <ul className={classes['cart-items']}>
 {cartCtx.items?.map((item) => (
        <CartItem
          key={item.id}
          name={item.name}
          amount={item.amount}
          price={item.price}
          onRemove={cartItemRemoveHandler.bind(null, item.id)}
          onAdd={cartItemAddHandler.bind(null, item)}
        />
      ))}
</ul>
);

  return (
    <Modal onClose={props.onClose} >
      {cartItems}
      <div className={classes.total}>
        <span>Total Amount</span>
        <span>{totalAmount}</span>
      </div>
      <div className={classes.actions}>
        <button className={classes['button--alt']}  onClick={props.onClose}>
            Close
        </button>
        {(cartCtx.items?.length > 0) && <button className={classes.button}>Order</button>}
      </div>
      {/* {console.log("amount", cartCtx.items)} */}
    </Modal>
  );
};

export default Cart;