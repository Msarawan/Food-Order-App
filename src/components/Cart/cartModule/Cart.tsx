import * as React from 'react';
import Modal from '../../UI/modal/Modal';
import CartItem from '../cartItem/CartItem';
import classes from './Cart.module.css';
//import CartContext from '../../../store/CartContext';
import { CartCtx } from '../../../store/CartContext';


interface ICartProps{
   onClose:()=>void
}

const Cart: React.FC<ICartProps> = ({onClose}) => {
  //const cartCtx = useContext(CartContext);
  const cartCtx = CartCtx();
  console.log(cartCtx.totalAmount)
  const totalAmount = `$${cartCtx.totalAmount.toFixed(2)}`;
  console.log(totalAmount)

  const cartItemRemoveHandler = (id:string) => {
    cartCtx.removeItem(id);
  };

  const cartItemAddHandler = (item:any) => {
    cartCtx.addItem({ ...item, amount: 1 });
};
   
 console.log(cartCtx.items);

const cartItems = (
    <ul className={classes['cart-items']}>
 {cartCtx.items?.map((item) => (
        <CartItem
          key={item['id']}
          name={item['name']}
          amount={item['amount']}
          price={item['price']}
          onRemove={cartItemRemoveHandler.bind(null, item['id'])}
          onAdd={cartItemAddHandler.bind(null, item)}
        />
      ))}
</ul>
);

  return (
    <Modal onClose={onClose} >
      {cartItems}
      <div className={classes.total}>
        <span>Total Amount</span>
        <span>{totalAmount}</span>
      </div>
      <div className={classes.actions}>
        <button className={classes['button--alt']}  onClick={onClose}>
            Close
        </button>
        {(cartCtx.items?.length > 0) && <button className={classes.button}>Order</button>}
      </div>
      {/* {console.log("amount", cartCtx.items)} */}
    </Modal>
  );
};

export default Cart;