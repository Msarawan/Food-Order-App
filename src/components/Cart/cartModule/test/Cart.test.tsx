import * as React from 'react';
 import {shallow} from 'enzyme';
 import   * as CartContext from '../../../../store/CartContext';
 import Cart from "../Cart";

jest.mock('../../cartItem/CartItem');
jest.mock('../../../UI/modal/Modal');


 describe('mock the component', () => {
 let wrapper:any;
 test('it should mock the context', () => {
     
  const cartCtx:any = {
     items:[{
       id : 1,
       amount:4,
       name:'aloo kulcha',
       price: 18.48,  
    }],
    addItem: jest.fn(),
    removeItem: jest.fn(),
    totalAmount: 18.45,
   }
   jest
      .spyOn(CartContext, 'CartCtx')
      .mockImplementation(() =>(cartCtx));
       wrapper = shallow(<Cart onClose={undefined}/>);
       console.log(wrapper.debug());
       //console.log(cartCtx.items);
  
    expect(wrapper).toMatchSnapshot(); 


});
  
});
