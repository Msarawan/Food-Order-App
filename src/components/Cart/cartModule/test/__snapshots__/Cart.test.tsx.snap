// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock the component it should mock the context 1`] = `
<MockedModal
  onClose={[Function]}
>
  <ul
    className="cart-items"
  >
    <MockedCartItem
      amount={4}
      key="1"
      name="aloo kulcha"
      onAdd={[Function]}
      onRemove={[Function]}
      price={18.48}
    />
  </ul>
  <div
    className="total"
  >
    <span>
      Total Amount
    </span>
    <span>
      $18.45
    </span>
  </div>
  <div
    className="actions"
  >
    <button
      className="button--alt"
      onClick={[Function]}
    >
      Close
    </button>
    <button
      className="button"
    >
      Order
    </button>
  </div>
</MockedModal>
`;
